controllers/upgrade_handlers.go

```diff
@@ -64,45 +64,11 @@ func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lc
 	// WARNING: the pod may not know if we are boot loop (for now)
 	if prePivot {
 		r.Log.Info("Starting pre pivot steps and will pivot to new stateroot with a reboot")
-		ctrlResult, err := r.prePivot(ctx, ibu)
-		if err != nil {
-			//todo: abort handler? e.g delete desired stateroot
-			utils.SetUpgradeStatusFailed(ibu, err.Error())
```

Comment:
The prePivot/postPivot functions are now the only places where Upgrade gets set to Failed, I guess? So we need to ensure all failure point in these functions do so. Not related to this PR, maybe, but I think it would be good to have that well-commented in these functions to make that requirement clear.

Reply:
Yes, all decisions(requeue+failures) should be made in one place which is inside of those functions. Will add comments in a further cleanup PR. 
As both completed and failed cases should stop the reconciliation (return doNotRequeue(), nil), with previous impelementation where the caller and called functions have mixed decisions, you cannot distingiush whether the result from called functions are failed or completed cases.